src/backend/mod.rs

```diff
@@ -0,0 +1,28 @@
+//! Backend (rendering/input) creation helpers
+//!
+//! Collection of common traits and implementation about
+//! rendering onto various targets and receiving input
+//! from various sources.
+//!
+//! Supported graphics backends:
+//!
+//! - glutin (headless/windowed)
+//!
+//! Supported input backends:
+//!
+//! - glutin (windowed)
+
+pub mod input;
+pub mod graphics;
+
+#[cfg(feature = "glutin")]
+pub mod glutin;
+
+#[cfg(feature = "glium")]
+mod glium;
+#[cfg(feature = "glium")]
+pub use glium::*;
+
+trait NewIdType {
+    fn new(id: u32) -> Self;
+}
```

Comment:
I'm not a huge fan of this trait, but I see the use. I've been using freestanding functions to do the same job in other projects, which is not really better.

Hopefully pub(restricted) will save us somewhere in the future.